swarm/network/discovery.go

```diff
@@ -145,9 +145,7 @@ func (d *discPeer) handleSubPeersMsg(msg *subPeersMsg) error {
 		})
 		log.Warn(fmt.Sprintf("found initial %v peers not farther than %v", len(peers), d.depth))
 		if len(peers) > 0 {
-			if err := d.Send(&peersMsg{Peers: peers}); err != nil {
-				return err
-			}
+			go d.Send(&peersMsg{Peers: peers})
```

Comment:
Looks like this needs a rebase on top of the latest `network-testing-framework` branch which already has this fix.

Reply:
wow I rebased on `network-testing-framework` already. I failed?

Reply:
Something has gone wrong, it is most likely commit `320ac1d` which is a merge commit. Pull requests shouldn't really have merge commits in if they have been rebased.

For the record, I very rarely run `git merge`, most of the time `git fetch` and `git rebase`.

You may also find the following `git log` command useful to see what is going on:

```
$ git log --graph --all --abbrev-commit --pretty=format:'%h -%d %s (%cr) %an'

* d81de06 - (ethersphere/pss-whisper-protocols) swarm/pss: Refactor PssProtocol to use encryption (20 hours ago) lash
*   320ac1d - (ethersphere/pss-whisper-noaddr) Merge branch 'pss-whisper-noaddr' of github.com:ethersphere/go-ethereum into pss-whisper-noaddr (22 hours ago) lash
|\  
| * ed939db - swarm/pss: Add API method for manual symkey add (27 hours ago) lash
| * 8cc5bec - swarm/pss: Fix asymmetric send test (6 days ago) lash
| * 3fd0d48 - swarm/pss: Add test variable addr length handshakes (7 days ago) lash
| * 72c36b1 - swarm/pss: Add symkey send test (8 days ago) lash
| * 9aed7aa - swarm/pss, whisper: Complete handshake (8 days ago) lash
| * 8deec11 - swarm/pss: Modify routing algo (13 days ago) lash
* | 81a942a - swarm/pss, whisper: partial address routing (22 hours ago) lash
* | 8e7d1d4 - (HEAD -> network-testing-framework, ethersphere/network-testing-framework) swarm/network: Prevent race in discovery simulation tests (2 days ago) Lewis Marshall
* | cdd5f91 - swarm/network: Make handleSubPeersMsg response async (3 days ago) Lewis Marshall
* |   0c5859e - Merge pull request #117 from ethersphere/pss-whisper (7 days ago) Viktor Trón
|\ \  
| * | d131687 - (ethersphere/pss-whisper) swarm/pss: Modify routing algo (13 days ago) lash
| |/  
| * 6e92fd1 - swarm/pss, whisper: granular partial address ctrl (13 days ago) lash

```

So merge commit `320ac1d` merges two histories, only one of which contains the latest `network-testing-framework` commits.